.github/workflows/main.yml

```
@@ -146,3 +146,41 @@ jobs:
           cat test_*.log || true
           cat valgrind_test.log || true
           cat gcc_errors_*.log || true
+
+  CMake:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-18.04, ubuntu-20.04 ]
+        build_type: [ '', -DCMAKE_BUILD_TYPE=Debug, -DCMAKE_BUILD_TYPE=Release, -DCMAKE_BUILD_TYPE=RelWithDebInfo, -DCMAKE_BUILD_TYPE=MinSizeRel ]
+        cc: [ clang, gcc ]
+        config:
+          # Static library build
+          - { CMAKEOPTIONS: '',                      }
+          # Shared library build
+          - { CMAKEOPTIONS: '-DBUILD_SHARED_LIBS=On' }
```

Let's not depend on the cmake default and set it explicitly here.
```suggestion
          - { CMAKEOPTIONS: '-DBUILD_SHARED_LIBS=OFF' }
          # Shared library build
          - { CMAKEOPTIONS: '-DBUILD_SHARED_LIBS=ON' }
```